OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/preferencecenter/OptimovePreferenceCenter.java

```diff
@@ -0,0 +1,300 @@
+package com.optimove.android.preferencecenter;
+
+import android.os.Handler;
+import android.os.Looper;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import com.optimove.android.OptimoveConfig;
+import com.optimove.android.main.tools.networking.HttpClient;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import okhttp3.Response;
+
+import java.io.IOException;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+
+public class OptimovePreferenceCenter {
+    private static final String TAG = OptimovePreferenceCenter.class.getName();
+    private static OptimovePreferenceCenter shared;
+    private static OptimoveConfig config;
+    private static @Nullable String customerId;
+    private static int tenantId;
+
+    /**
+     * package
+     */
+    static ExecutorService executorService;
+    /**
+     * package
+     */
+    static final Handler handler = new Handler(Looper.getMainLooper());
+
+    public interface PreferencesGetHandler {
+        void run(@Nullable Preferences preferences);
+    }
+
+    public interface PreferencesSetHandler {
+        void run(Boolean result);
+    }
+
+    public enum Channel {
+        MOBILE_PUSH(489),
+        WEB_PUSH(490),
+        SMS(493);
+        private final int channel;
+
+        Channel(int channel) {
+            this.channel = channel;
+        }
+
+        @NonNull
+        public int getValue() {
+            return channel;
+        }
+
+        public static Channel getChannelByValue ( int value) {
+            switch (value) {
+                case 489:
+                    return MOBILE_PUSH;
+                case 490:
+                    return WEB_PUSH;
+                case 493:
+                    return SMS;
+                default:
+                    throw new IllegalArgumentException("Preference center does not support channel " + value);
+            }
+        }
+    }
+
+    public static OptimovePreferenceCenter getInstance() {
+        if (shared == null) {
+            throw new IllegalStateException("OptimovePreferenceCenter is not initialized");
+        }
+        return shared;
+    }
+
+    /**
+     * Initializes an instance of OptimovePreferenceCenter
+     *
+     * @param currentConfig current config
+     * @param currentCustomerId current customer id
+     * @param currentTenantId current tenant id
+     */
+    public static void initialize(OptimoveConfig currentConfig, String currentCustomerId, int currentTenantId) {
+        if (currentConfig.getBrandGroupId() == null) {
+            Log.d(TAG, "Preference center has not been enabled");
+            return;
```

Comment:
are we going to log this for everyone who hasnt set brand group id now? can change `isPCConfigured` to check brand group id is set

Reply:
in fact, maybe even possible to actually add PC to the feature set? 

then, for example, if a customer comes that doesnt have brandGroupId during initialisation, we can make brandGroupId argument nullable. but when you call enablePreferenceCenter, it adds it to the feature set